app/controllers/teams_controller.rb

```diff
@@ -5,8 +5,9 @@ class TeamsController < ApplicationController
   before_action :ensure_leader, only: [:update, :delete]
 
   def ensure_valid_team_id
-    @team = Team.find(params[:id])
-    if @team == nil
+    begin
+      @team = Team.find(params[:id])
+    rescue
```

Comment:
Rescue is a better alternative. I do feel this method is quite similar to other ensure methods and might be able to be refactored.

Reply:
noted, will consider for next pr